src/platform/silabs/rs911x/BLEManagerImpl.cpp

```diff
@@ -88,10 +99,13 @@ void sl_ble_init()
                                     rsi_ble_on_event_indication_confirmation, NULL);
 
     //  Exchange of GATT info with BLE stack
+
     rsi_ble_add_matter_service();
 
     //  initializing the application events map
     rsi_ble_app_init_events();
+    sl_get_bluethooth_addr(addr, BLE_ADDR_LENGTH);
```

Comment:
This function call should be removed completely. It's unsafe and there is ways to make it work in a more elegant way IMO. something like

```
uint64_t randomAddr = chip::Crypto::GetRandU64();
uint8_t randomAddrRSI[6] = {0};
memcpy(randomAddrRSI, &randomAddr, 6);